### DISCUSSION TITLE
Disable clangd if compile_commands.json is not found

### DISCUSSION THREAD

**Discussion by steve-lorimer** [VonHeikemen/lsp-zero.nvim#262] [Q&A]
90% of my work is done in a project for which I've generated `compile_commands.json`, so clangd can provide proper diagnostics etc.

However, if I switch to another project which doesn't have `compile_commands.json`, I just get tonnes of errors everywhere, because clangd can't resolve any of the includes or compiler flags etc. I then have to run `:LspStop` to hide all the cruft. If I open another file, the same thing happens until I turn lsp off, and so on.

How can I configure lsp/clangd to only enable if it finds `compile_commands.json`?

**Comment by VonHeikemen** [comment]
I think this should work.

```lua
local lsp = require('lsp-zero')

lsp.on_attach(function(client, bufnr)
  lsp.default_keymaps({buffer = bufrn})
end)

local lspconfig = require('lspconfig')

lspconfig.clangd.setup({
  single_file_support = false,
  root_dir = lspconfig.util.root_pattern('compile_commands.json')
})

lsp.setup()
```

**Reply by steve-lorimer** [reply]
That's so easy, thank you!

Another related question please.

We use a non-standard build directory, which means `compile_commands.json` is in an unusual location, `<project>/.build/debug/compile_commands.json`. 

How can I configure clangd to look in the usual locations, and also the above locaiton?

**Reply by VonHeikemen** [reply]
Try this with the root pattern function.

```lua
lspconfig.util.root_pattern(
    '.build/debug/compile_commands.json',
    'compile_commands.json',
)
```

**Reply by steve-lorimer** [reply]
Thank you! I'll give it a try
